miraclelinux/meta-debian-extended
# isc-dhcp: Update version from 4.4.1-2+deb10u2 to 4.4.1-2+deb10u3

# Purpose of pull request
Debian released a security update for isc-dhcp package.
DLA: DLA 3326-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemu image
3. Check initial IP address at qemu startup
4. Add settings to /etc/dhcp/dhcpd.conf and restart dhcpd service
5. Run dhclient command and check assigned IP address 
6. Check /var/log/messages


### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
IMAGE_INSTALL_append = " dhcp-client dhcp-server"
IMAGE_INSTALL_append = " coreutils"
```
```
$ bitbake core-image-minimal
```

### Run qemu image
```
$ runqemu qemuarm64 nographic
```
### Check initial IP address at qemu startup
```
ip address
```
### Add settings to /etc/dhcp/dhcpd.conf and restart dhcpd service
Add settings to /etc/dhcp/dhcpd.conf.
```
subnet 192.168.7.0 netmask 255.255.255.0 {
  option routers                192.168.7.1;
  option subnet-mask            255.255.255.0;
  option domain-name-servers    8.8.8.8;
  option time-offset            -18000;
  range 192.168.7.100 192.168.7.200;
}
```
Restart dhcpd service.
```
/etc/init.d/dhcp-server restart
```

### Run dhclient command and check assigned IP address
```
dhclient eth0
ip address
```
### Check /var/log/messages
```
tail -n 15 /var/log/messages
```


## Test result
Build succeeded.
```
$ bitbake dhcp
Loading cache: 100% |########################################################################################################################################| Time: 0:00:00
Loaded 2373 entries from dependency cache.
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |######################################################################################################################################| Time: 0:00:14
Parsing of 1353 .bb files complete (843 cached, 510 parsed). 2373 targets, 80 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.6"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:8f17d6cf6448ceaac5c7f35c538b91d477f79424"
meta-debian-extended = "update-isc-dhcp_4.4.1-2+deb10u3:b9d31e247312b52e2c3a66146f9c095c87b686d5"
meta-emlinux         = "HEAD:74bfbdbfef98aa830d3f30dfbff07b134b8da9df"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |###################################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 286 Found 276 Missed 10 Current 148 (96% match, 97% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 1995 tasks of which 1966 didn't need to be rerun and all succeeded.
```

Check initial IP address at qemu startup
```
root@qemuarm64:~# ip address
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast qlen 1000
    link/ether 52:54:00:12:34:04 brd ff:ff:ff:ff:ff:ff
    inet 192.168.7.4/24 brd 192.168.7.255 scope global eth0
       valid_lft forever preferred_lft forever
```

Add settings to /etc/dhcp/dhcpd.conf and restart dhcpd service
```
root@qemuarm64:~# tail /etc/dhcp/dhcpd.conf
#    range 10.0.29.10 10.0.29.230;
#  }
#}
subnet 192.168.7.0 netmask 255.255.255.0 {
  option routers                192.168.7.1;
  option subnet-mask            255.255.255.0;                        
  option domain-name-servers    8.8.8.8;                
  option time-offset            -18000;
  range 192.168.7.100 192.168.7.200;
}
root@qemuarm64:~# /etc/init.d/dhcp-server restart
Stopping DHCP server: dhcpd3no /usr/sbin/dhcpd found; none killed
.
Starting DHCP server: .
```
Run dhclient command and check assigned IP address 
```
root@qemuarm64:~# dhclient eth0
root@qemuarm64:~# ip address
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast qlen 1000
    link/ether 52:54:00:12:34:04 brd ff:ff:ff:ff:ff:ff
    inet 192.168.7.4/24 brd 192.168.7.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet 192.168.7.100/24 brd 192.168.7.255 scope global secondary eth0
       valid_lft forever preferred_lft forever
```
Check /var/log/messages
```
root@qemuarm64:~# tail -n 15 /var/log/messages
Mar  6 08:05:32 qemuarm64 local7.info dhcpd: Internet Systems Consortium DHCP Server 4.4.1
Mar  6 08:05:32 qemuarm64 local7.info dhcpd: Copyright 2004-2018 Internet Systems Consortium.
Mar  6 08:05:32 qemuarm64 local7.info dhcpd: All rights reserved.
Mar  6 08:05:32 qemuarm64 local7.info dhcpd: For info, please visit https://www.isc.org/software/dhcp/
Mar  6 08:05:32 qemuarm64 local7.info dhcpd: Wrote 0 leases to leases file.
Mar  6 08:05:32 qemuarm64 local7.info dhcpd: Server starting service.
Mar  6 08:07:15 qemuarm64 daemon.info dhclient: DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 6
Mar  6 08:07:16 qemuarm64 local7.info dhcpd: DHCPDISCOVER from 52:54:00:12:34:04 via eth0
Mar  6 08:07:17 qemuarm64 local7.info dhcpd: DHCPOFFER on 192.168.7.100 to 52:54:00:12:34:04 via eth0
Mar  6 08:07:17 qemuarm64 daemon.info dhclient: DHCPOFFER of 192.168.7.100 from 192.168.7.4
Mar  6 08:07:17 qemuarm64 daemon.info dhclient: DHCPREQUEST for 192.168.7.100 on eth0 to 255.255.255.255 port 67
Mar  6 08:07:17 qemuarm64 local7.info dhcpd: DHCPREQUEST for 192.168.7.100 (192.168.7.4) from 52:54:00:12:34:04 via eth0
Mar  6 08:07:17 qemuarm64 local7.info dhcpd: DHCPACK on 192.168.7.100 to 52:54:00:12:34:04 via eth0
Mar  6 08:07:17 qemuarm64 daemon.info dhclient: DHCPACK of 192.168.7.100 from 192.168.7.4
Mar  6 08:07:17 qemuarm64 daemon.info dhclient: bound to 192.168.7.100 -- renewal in 232 seconds.
```
